core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1405,6 +1405,8 @@ public synchronized Sha256Hash hashForWitnessSignature(
             bos.write(hashSequence);
             bos.write(inputs.get(inputIndex).getOutpoint().getHash().getReversedBytes());
             uint32ToByteStreamLE(inputs.get(inputIndex).getOutpoint().getIndex(), bos);
+            VarInt scriptLength = new VarInt(scriptCode.length);
```

Comment:
The last time I looked at this, I didn't see the error because BIP143 just states "serialized as scripts inside CTxOuts".

Now I agree. There is even a [TODO in `TransactionOutput.java`](https://github.com/bitcoinj/bitcoinj/blob/f94f8cd692144c5423950b9063db7e7ecd8a89c4/core/src/main/java/org/bitcoinj/core/TransactionOutput.java#L151) to move the script serialization to `Script.java`. Which would perhaps helped prevent the mistake.